### PR TITLE
fix(core): fix busy indicator accessibility bugs

### LINKED ISSUE #6095: Bug: Accessibility Testing_Core_Busy Indicator
Accessibility: Busy Indicator-Core Component (There are couple of accessibility issues found while testing, so I have attached error document for you reference.)
[Busy Indicator_Core_Ariba Controls.docx](https://github.com/SAP/fundamental-ngx/files/6877032/Busy.Indicator_Core_Ariba.Controls.docx)

Busy Indicator- Core

1.1	Message Details

Raise Ticket	
Category	
Preprocessed By	
Description	

1.2 Application & Test Environment Details

Application Name:	ARIBA Controls- Fundamental Library for Angular
URL:	https://fundamental-ngx.netlify.app/#/core/busyIndicator

Technology:	Angular 11
Version: 	V 0.31.0-rc.125
Web browser:	Google Chrome 92.0.4515.107 (Official Build) (64-bit)
Operating System:	Microsoft Windows 10 x64
Screen Reader:	JAWS 2021.2105.53 ILM

 

Issue 1

Priority:	Medium
Checkpoint:	ACC-264

(264.1)

•	Search ‘Busy Indicator’ from the left panel

Observed Behavior:

On accessing the ‘Default Busy indicator’, the percentage of loading is not conveyed.

Expected Behavior:
The percentage of loading information should be conveyed so that user will aware of the progress in loading.

Screenshot:

 
image

Issue 2

Priority:	Medium
Checkpoint:	ACC-264

(264.1)

•	Search ‘Busy Indicator’ from the left panel

Observed Behavior:

When ‘focus’ is on the complete wrappers, reading is made as ‘loading’ and the labels ‘use Busy indicator as a block wrapper’ and ‘use Busy Indicator as a inline wrapper’ are not conveyed.

Expected Behavior:
All the labels should get associated with the corresponding controls when accessed.
Screenshot:

 
image

Note: The issue 2 persists with Jaws 2019 as well.

### PR DESCRIPTION
**Pull request by aberikashvili**
#### Please provide a link to the associated issue.
Closes #6095

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [N/A] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [N/A] update `README.md`
- [N/A] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Review by N1XUS — APPROVED** [review]

File: libs/core/src/lib/busy-indicator/busy-indicator.component.ts [hunk 1]
```diff
@@ -58,6 +61,15 @@ export class BusyIndicatorComponent {
     @Input()
     ariaLive: 'assertive' | 'polite' | null = 'polite';
 
+    @Input()
```
Comment (platon-rov) [inline libs/core/src/lib/busy-indicator/busy-indicator.component.ts:64]: Add TypeDoc comments please.

**Review by platon-rov — CHANGES_REQUESTED** [review]
About this one https://github.com/SAP/fundamental-ngx/issues/6095#issuecomment-906184020

Seems like it's not working properly, I cannot jump to the inner content of the Busy Indicator (table in my case).

Screenshot 2021-09-08 at 15 08 32

File: libs/core/src/lib/busy-indicator/busy-indicator.component.ts (hunk 1 above)
Comment (InnaAtanasova) [reply libs/core/src/lib/busy-indicator/busy-indicator.component.ts:64]: +1

File: libs/core/src/lib/busy-indicator/busy-indicator.component.ts [hunk 2]
```diff
@@ -24,6 +24,9 @@ export type BusyIndicatorSize = 's' | 'm' | 'l';
         '[attr.aria-busy]': 'loading',
         '[attr.aria-live]': 'ariaLive',
         '[attr.aria-label]': 'ariaLabel',
+        '[attr.aria-valuemin]': 'ariaValueMin',
+        '[attr.aria-valuemax]': 'ariaValueMax',
+        '[attr.aria-valuenow]': 'ariaValueNow',
```
Comment (InnaAtanasova) [inline libs/core/src/lib/busy-indicator/busy-indicator.component.ts:29]: Can you explain me how are these related to the a11y and the component? As per this documentation: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-valuemax_attribute

these values are for a range widget such as a slider, spinbutton or progressbar.

Comment (aberikashvili) [reply libs/core/src/lib/busy-indicator/busy-indicator.component.ts:29]: @InnaAtanasova per the issue requirements, please see the issue description https://github.com/SAP/fundamental-ngx/issues/6095
image

Comment (InnaAtanasova) [reply libs/core/src/lib/busy-indicator/busy-indicator.component.ts:29]: This is a busy indicator, not a progress bar.

File: libs/core/src/lib/busy-indicator/busy-indicator.component.ts (hunk 1 above)
Comment (aberikashvili) [reply libs/core/src/lib/busy-indicator/busy-indicator.component.ts:65]: done

**Comment by aberikashvili** [comment]
> About this one [#6095 (comment)](https://github.com/SAP/fundamental-ngx/issues/6095#issuecomment-906184020)
> 
> Seems like it's not working properly, I cannot jump to the inner content of the Busy Indicator (table in my case).
> 
> Screenshot 2021-09-08 at 15 08 32

@platon-rov I addressed your comments, please check again

**Review by droshev — CHANGES_REQUESTED** [review]
I think there was a confusion. This is not a progress indicator. This is a busy indicator where the progress is indeterminate. Actually we won't need to do anything here.

**Comment by aberikashvili** [comment]
Per @droshev 's [note](https://github.com/SAP/fundamental-ngx/pull/6472#pullrequestreview-751306259), I removed aria attributes from busy-indicator

**Review by Betrozov — APPROVED** [review]

**Review by mikerodonnell89 — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]

**Comment by platon-rov** [comment]
@aberikashvili great, now I see the busy indicator as the wrapper doesn't interrupt the screen-reader. But for some reason focus is present for that wrapper, could you look into it? Example: https://deploy-preview-6472--fundamental-ngx.netlify.app/#/platform/table

image

And there is no focus present without your changes: https://fundamental-ngx.netlify.app/#/platform/table

**Comment by aberikashvili** [comment]
@platon-rov I've addressed your latest comment, please check again

**Comment by platon-rov** [comment]
Tab stop still present, here are the stops how it was and how it should be. Basically, the busy container shouldn't add the tab stop.

Screenshot 2021-09-21 at 13 13 41

**Review by platon-rov — APPROVED** [review]
great work!

**Comment by platon-rov** [comment]
Please fix travis

**Review by droshev — APPROVED** [review]

**Comment by platon-rov** [comment]
@aberikashvili what's the status of this pr? Should qa review to be started?

**Comment by aberikashvili** [comment]
> @aberikashvili what's the status of this pr? Should qa review to be started?

I think yes, PR is ready to merge

**Pull request merged**